### PR TITLE
vSphere - remove hard defined vlans for zones

### PR DESCRIPTION
**Pull request by WenXinWei**

### CODE REVIEW

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by WenXinWei** [comment]
@jianlinliu please help review when you have time. cc @jcpowermac

File: ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh [hunk 1]
```diff
@@ -116,12 +115,8 @@ source "${SHARED_DIR}/govc.sh"
 
 DATACENTERS=("$GOVC_DATACENTER")
 # If testing a zonal install, there are multiple datacenters that will need to be cleaned up
-# vlanid between 1287 and 1302 will use profile:vsphere-multizone-2
-if [ ${vlanid} -ge 1287 ] && [ ${vlanid} -le 1302 ]; then
-  DATACENTERS=(
-    "IBMCloud"
-    "datacenter-2"
-  )
+if [[ $JOB_NAME_SAFE == *"zones"* ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh:118]: The same comment as https://github.com/openshift/release/pull/46163#discussion_r1413309957

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

File: ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh [hunk 2]
```diff
@@ -116,12 +115,8 @@ source "${SHARED_DIR}/govc.sh"
 
 DATACENTERS=("$GOVC_DATACENTER")
 # If testing a zonal install, there are multiple datacenters that will need to be cleaned up
-# vlanid between 1287 and 1302 will use profile:vsphere-multizone-2
-if [ ${vlanid} -ge 1287 ] && [ ${vlanid} -le 1302 ]; then
-  DATACENTERS=(
-    "IBMCloud"
-    "datacenter-2"
-  )
+if [[ "${vsphere_datacenters[*]}" -gt 0 ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh:118]:
```suggestion
if [[ "${#vsphere_datacenters[@]}" -gt 0 ]]; then
```

Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh]: why add '#' here before the array? what's the meaning here?

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh]: we are trying to get the length of `vsphere_datacenters` array to determinate if the cluster profile has multiple datacenters.

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh]: `#` can be used to get the length of the array.

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-serial

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-static-ovn

File: ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh [hunk 3]
```diff
@@ -10,6 +10,11 @@ if [[ -z "${LEASED_RESOURCE}" ]]; then
 fi
 
 declare vsphere_datacenter
+
+# only used in zonal and vsphere environments with
+# multiple datacenters
+
+declare vsphere_datacenters
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh:17]: Per the test job [log](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/49193/rehearse-49193-periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-serial/1763118495857381376), we need some update.
```suggestion
declare -a vsphere_datacenters
vsphere_datacenters=()
```

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by jianlinliu** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-ovn-serial

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-static-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28 periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by jianlinliu** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.13-e2e-vsphere-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-e2e-vsphere-zones

**Pull request merged**